src/main/java/org/opentripplanner/traffic/SegmentSpeedSample.java

```diff
@@ -128,10 +128,14 @@ public SegmentSpeedSample(SummaryStatistics stats) {
     /** create a speed sample using a function */
     public SegmentSpeedSample(double averageSpeed, double[] hourBins) {
         this.average = encodeSpeed(averageSpeed);
-        this.hourBins = new short[hourBins.length];
-
-        for (int i = 0; i < hourBins.length; i++) {
-            this.hourBins[i] = Double.isNaN(hourBins[i]) ? this.average : encodeSpeed(hourBins[i]);
+        if (hourBins != null){
```

Comment:
fix formatting: space before {

Reply:
On reflection, though: is hourBins ever null?  Should we throw an exception if it is?

Reply:
On line 36 it is considered a proper value.
So throwing an exception is not appropriate.